pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/RoundRobinScheduler.java

```diff
@@ -229,6 +230,41 @@ public OpChain next(long time, TimeUnit timeUnit) throws InterruptedException {
     return _ready.poll(time, timeUnit);
   }
 
+  @Override
+  public void cancel(OpChainId opChainId, Future<?> future) {
+    if (!_aliveChains.containsKey(opChainId)) {
+      return;
+    }
+    _lock.lock();
+    try {
+      if (!_aliveChains.containsKey(opChainId)) {
+        return;
+      }
+      // First mark OpChain as interrupted to ensure it throws next time it is scheduled/run
+      OpChain opChain = _aliveChains.get(opChainId);
+      opChain.markInterrupted();
+      if (_available.containsKey(opChainId)) {
+        _available.remove(opChainId);
+        _ready.offer(opChain);
+      } else {
+        // OpChain is either running or in ready queue. We use the Future to cancel which will interrupt the thread
```

Comment:
Edit: If the task hadn't started running yet then it won't be run again. This needs to be fixed